### DISCUSSION TITLE
Anonymous access to read files - how to switch off listing of files/directories?

### DISCUSSION THREAD

**Discussion by ser** [minio/minio#18451] [Q&A] [answered]
I have turned on Anonymous access into a bucket:

![image](https://github.com/minio/minio/assets/231537/545ee9d8-6045-488d-a6b1-dbee5a368e52)

But how can I turn off listing of files and directories for anonymous users? I want them to be allowed to access but not list whole content.

**Comment by danielcommesse** [comment, marked as answer]
Go to `Summary` > `Access Policy` and click the edit button.

Select `Custom` from the dropdown and enter the following policy.

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": [
                    "*"
                ]
            },
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::your_bucket_name/*"
            ]
        }
    ]
}
```

This will allow to only view/download files, but no listing.

Settings anonymous to `download/public` will allow listing, there is no way to allow `download-but-deny-listing` via anonymous, you have to use a custom policy.

Hope it helps!
